### PR TITLE
Lint metadata

### PR DESCRIPTION
**Pull request by arturo-seijas**
Lint metadata.yaml as part of the test stage

### CODE REVIEW

**Review by gtrkiller — DISMISSED** [review]
LGTM

File: .github/files/metadata.schema [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    "type" : "object",
+    "properties" : {
+        "name" : {"type" : "string"},
+        "display-name" : {"type" : "string"},
+        "summary" : {"type" : "string", "maxLength": 50},
+        "description" : {"type" : "string"},
+        "docs" : {"type" : "string"},
+        "maintainers" : {"type" : "array", "items": {"type": "string", "enum": ["launchpad.net/~canonical-is-devops"]}},
+        "issues" : {"type" : "string"},
+        "tags" : {"type" : "array", "items": {"type": "string"}}
```
Comment (amandahla) [inline .github/files/metadata.schema:11]: Would be possible to check for "assumes" too?
```yaml
description: Description
assumes:
  - k8s-api

Comment (arturo-seijas) [reply .github/files/metadata.schema:11]: The workflows also support machine charms, so I don't think it's a good idea

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -120,6 +120,17 @@ jobs:
...
           dockerfile: "${{ env.Dockerfiles }}"
+  lint-metadata:
+    name: Lint metadata.yaml
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install check-jsonschema
+        run: python3 -m pip install check-jsonschema
+      - name: Download schema
+        run: curl -o $PWD/metadata.schema https://raw.githubusercontent.com/canonical/operator-workflows/main/.github/files/metadata.schema
```
Comment (weiiwang01) [inline .github/workflows/test.yaml:131]: Add `-sfSL` to curl? `-s`: silence the download progress; `-S`: print error in silence mode; `-f` fail on server errors; `-L` follow redirects.

Comment (arturo-seijas) [reply .github/workflows/test.yaml]: Done, thx

File: .github/files/metadata.schema [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+    "type" : "object",
+    "properties" : {
+        "name" : {"type" : "string"},
+        "display-name" : {"type" : "string"},
+        "summary" : {"type" : "string", "maxLength": 50},
+        "description" : {"type" : "string"},
+        "docs" : {"type" : "string"},
+        "maintainers" : {"type" : "array", "items": {"type": "string", "enum": ["launchpad.net/~canonical-is-devops"]}},
```
[the remark this answers is not in the document]
Comment (arturo-seijas) [reply .github/files/metadata.schema:9]: Good point, maybe we should only check the field is populated, thoughts?

[the remark this answers is not in the document]
Comment (weiiwang01) [reply .github/files/metadata.schema:9]: Yes, that's a good idea.

**Review by weiiwang01 — DISMISSED** [review]

File: .github/files/metadata.schema (hunk 3 above)
[the remark this answers is not in the document]
Comment (arturo-seijas) [reply .github/files/metadata.schema]: Done

**Review by weiiwang01 — APPROVED** [review]

**Review by gtrkiller — APPROVED** [review]
LGTM

**Comment by jdkandersson** [comment]
The test failed I think, I'll change the repo config top require status checks to pass

**Pull request merged**